### ISSUE TITLE
setOnSpinnerOutsideTouchListener does not fire

### ISSUE THREAD

**Issue by davischung** [skydoves/PowerSpinner#11]
**Please complete the following information:**
- Library Version [e.g. v1.0.0] : 1.0.5
- Affected Device(s) : Google Pixel 3a
 
**Describe the Bug:**

Im using Kotlin with PowerSpinner, setting the onSpinnerOutsideTouchListener using `setOnSpinnerOutsideTouchListener { view, motionEvent -> }` does not get fired while touching outside spinner when the spinner is open

**Expected Behavior:**

Expect to get event fired when touching outside spinner when the spinner is open, so that I could close the spinner. It would be good if the spinner could be closed when user click the physical back button.

**Comment by skydoves** [comment]
@davischung 
Hi, I checked it and It will be fixed in the next release.
Thank you for your issue :)

**Comment by mariooooo** [comment]
it would be great if you fixed it

**Comment by skydoves** [comment]
If there is no more issue, this fixed version will be released today. Thank you :)

**Comment by skydoves** [comment]
@davischung @mariooooo 
It's released a new version [1.0.6](https://github.com/skydoves/PowerSpinner/releases/tag/1.0.6). Thank you for your issue :)

**Comment by davischung** [comment]
@skydoves @mariooooo 
Verified it fired on 1.0.6, good job.

**Comment by mariooooo** [comment]
@skydoves 
I don’t know, but it doesn’t work for me, maybe I'm doing something
`spinner.setOnSpinnerOutsideTouchListener { view, motionEvent ->
            println("outside touch")
            spinner.dismiss()
        }`
[youtube](https://youtu.be/RdS2GJShLgU)

**Comment by Elytum** [comment]
Hi !
Same here, 1.0.6 didn't fix the issue, still having to resort to a dirty use of "dispatchTouchEvent" in the activity. 
Will try to understand how to fix it later if needed, anyway thanks for the library !

**Comment by mariooooo** [comment]
@skydoves 
what about this problem?

**Comment by skydoves** [comment]
@mariooooo 
Hi, did you build using [the demo project](https://github.com/skydoves/PowerSpinner/tree/master/app)?
I would not be fired if the spinnerVIew is not `focusable` because of other editText.

Add the below code to the MainActivity.
```kotlin
spinnerView.spinnerOutsideTouchListener = object : OnSpinnerOutsideTouchListener {
      override fun onSpinnerOutsideTouch(view: View, event: MotionEvent) {
        Toast.makeText(baseContext, "touched outside", Toast.LENGTH_SHORT).show()
        spinnerView.dismiss()
      }
    }
```

**Comment by davischung** [comment]
@skydoves 
There is a side-issue produced by this `setOnSpinnerItemSelectedListener` fix. If you click on the `PowerSpinnerView` it will shows the dropdown, clicking one more time should dismiss the dropdown. 

The intention to add a `setOnSpinnerOutsideTouchListener` is to dismiss the dropdown while click was being detected outside the dropdown, to implement is to call `dismiss()` in the callback function. However, by adding a `setOnSpinnerOutsideTouchListener`, clicking on the `PowerSpinnerView` when dropdown is showing will fire `showOrDismiss` twice if `dismiss()` is being called within `setOnSpinnerOutsideTouchListener {}`, resulting unable to dismiss the `PowerSpinnerView` dropdown by clicking it again.

One quick fix on my side is to add some handling like below to override the `onClickListener`:
```
var dismissTime: Date? = null
binding.spinnerRegionCode.apply {
    ...
    setOnSpinnerOutsideTouchListener { _, _ ->
        dismissTime = Date()
        binding.spinnerRegionCode.dismiss()
    }
}
binding.spinnerRegionCode.setOnClickListener {
    if (Date().time - (dismissTime?.time ?: 0) > 200) {
        binding.spinnerRegionCode.showOrDismiss()
    }
    dismissTime = null
}
```

See if you could handle it within the library. Thanks.

**Comment by ShivamDev31** [comment]
Facing the same issue as @davischung

**Comment by skydoves** [comment]
It is released on the new version `1.0.8`!
And you can set the new attribute `debounce_duration`.
